src/wp-cron.php

```diff
@@ -18,19 +18,11 @@
 
 ignore_user_abort( true );
 
-if ( ! headers_sent() ) {
-	header( 'Expires: Wed, 11 Jan 1984 05:00:00 GMT' );
-	header( 'Cache-Control: no-cache, must-revalidate, max-age=0' );
-}
-
-// Don't run cron until the request finishes, if possible.
-if ( function_exists( 'fastcgi_finish_request' ) ) {
-	fastcgi_finish_request();
-} elseif ( function_exists( 'litespeed_finish_request' ) ) {
-	litespeed_finish_request();
-}
-
 if ( ! empty( $_POST ) || defined( 'DOING_AJAX' ) || defined( 'DOING_CRON' ) ) {
+	if ( ! headers_sent() ) {
+		header( 'Expires: Wed, 11 Jan 1984 05:00:00 GMT' );
+		header( 'Cache-Control: no-cache, must-revalidate, max-age=0' );
```

Comment:
Just noticed this is missing `no-store`:

```suggestion
		header( 'Cache-Control: no-store, no-cache, must-revalidate, max-age=0' );
```

Reply:
As I understand the intention is to prevent revere proxies from ever caching the response. This is what `no-store` (or `private`) would [achieve](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Cache-Control#:~:text=shared%20caches%20only.-,NO%2DSTORE,-The%20no%2Dstore):

> The `no-store` response directive indicates that any caches of any kind (private or shared) should not store this response.

Reply:
From memory the site health component uses a POST request to see if the endpoint is available. Again from memory, it prevents the site health screen triggering cron jobs twice, once as the admin screen loads and once as the loopback request is tested.

Reply:
Ah, ok. I guess that Site Health test would need to be disabled when the cron endpoint is disabled?